jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/serviceAccounts/PrincipleOfLeastPrivilegeRule.java

```diff
@@ -0,0 +1,128 @@
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, "Service accounts should not be given admin privilege to avoid to prevent any accidental or intentional modifications that may lead to data loss or data leaks");
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, "Service account does not have least privilege");
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========PrincipleOfLeastPrivilegeRule ended with annotation{} : =========", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("========PrincipleOfLeastPrivilegeRule ended with success=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+
+    }
+
+    private boolean checkIfServiceAccountHasLeastPrivilege(String vmEsURL, Map<String, Object> mustFilter) throws Exception {
+        JsonArray hitsJsonArray = GCPUtils.getHitsArrayFromEs(vmEsURL, mustFilter);
+        logger.debug("========PrincipleOfLeastPrivilegeRule started========= {}",hitsJsonArray);
+
+        if (hitsJsonArray.size() > 0) {
+            JsonObject serviceAccountObject = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                    .get(PacmanRuleConstants.SOURCE);
+
+            logger.debug("Validating the data item: {}", serviceAccountObject);
+            JsonObject rolesMembers = serviceAccountObject.getAsJsonObject()
+                    .get(PacmanRuleConstants.ROLES_MEMBERS).getAsJsonObject();
+          JsonArray editorMembers=  rolesMembers.getAsJsonArray("roles/editor");
+          JsonArray ownerMembers=rolesMembers.getAsJsonArray("roles/owner");
+          JsonArray adminMembers=rolesMembers.getAsJsonArray("roles/admin");
+          for(JsonElement editorMember:editorMembers)
+          {
+              if (editorMember.getAsString().startsWith("serviceAccount"))
+                  return false;
+          }
+            for(JsonElement ownerMember:ownerMembers)
+            {
+                if (ownerMember.getAsString().startsWith("serviceAccount"))
+                    return false;
+            }
+            for(JsonElement adminMember:adminMembers)
+            {
+                if (adminMember.getAsString().startsWith("serviceAccount"))
+                    return false;
+            }
+
```

Comment:
Policy should be validated for against membership of the target service account. Here the policy will fail for every service account if even one service account have editor/admin/owner role